src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js

```diff
@@ -0,0 +1,314 @@
+        expect(readAsText).toHaveBeenCalled();
+        expect(parentControls['input-field-name'].updateValueAndValidity).toHaveBeenCalledTimes(1);
+        expect(parent.meta.updateIncident).toHaveBeenCalled();
+      });
+
+      it('uploads already uploaded file and triggers multiple errors', async () => {
+        jest.useFakeTimers();
+        handler.mockImplementation(() => ({ value: [file2, file3, file4] }));
+        const { queryByTestId, findByTestId } = render(
+          withAppContext(
+            <FileInput
+              parent={parent}
+              handler={handler}
+              meta={{
+                ...metaFields,
+                maxNumberOfFiles: 3,
+                maxFileSize,
+                allowedFileTypes: ['image/jpeg'],
+              }}
+            />
+          )
+        );
+
+        const fileInputElement = queryByTestId('fileInputUpload');
+        act(() => {
+          fireEvent.change(fileInputElement, { target: { files: [file1] } });
+        });
+
+        await findByTestId('fileInputUpload');
+
+        expect(FileReader).toHaveBeenCalled();
+        expect(addEventListener).toHaveBeenCalledWith('load', expect.any(Function));
+        expect(readAsText).toHaveBeenCalled();
+        expect(parentControls['input-field-name'].updateValueAndValidity).toHaveBeenCalledTimes(2);
+        expect(parent.meta.updateIncident).toHaveBeenCalledWith({
+          'input-field-name_previews': expect.any(Array),
+          'input-field-name': [file4, file1],
+        });
+
+        jest.runAllTimers();
+      });
+
+      it('uploads a file with no extra validations', async () => {
+        handler.mockImplementation(() => ({ value: undefined }));
+
+        const { queryByTestId, findByTestId } = render(
+          withAppContext(
+            <FileInput
+              parent={parent}
+              handler={handler}
+              meta={{
+                ...metaFields,
+                maxNumberOfFiles: 3,
+                maxFileSize,
+                allowedFileTypes: ['image/jpeg'],
+              }}
+            />
+          )
+        );
+
+        const fileInputElement = queryByTestId('fileInputUpload');
+        act(() => {
+          fireEvent.change(fileInputElement, { target: { files: [file1] } });
+        });
+
+        await findByTestId('fileInputUpload');
+
+        expect(FileReader).toHaveBeenCalled();
+        expect(addEventListener).toHaveBeenCalledWith('load', expect.any(Function));
+        expect(readAsText).toHaveBeenCalled();
+        expect(parentControls['input-field-name'].updateValueAndValidity).toHaveBeenCalledTimes(1);
+        expect(parent.meta.updateIncident).toHaveBeenCalledWith({
+          'input-field-name_previews': expect.any(Array),
+          'input-field-name': [file1],
+        });
+      });
+
+      it('it removes 1 file when remove button was clicked', async () => {
+        parent.value = {
```

Comment:
I would strongly advise not to rewrite variables, but to destructure them as prop:
```jsx
<FileInput
  parent={{...parent, value: { ... } }}
  ...
/>
```

Reply:
Sure!